engine/src/main/java/io/zeebe/engine/metrics/StreamProcessorMetrics.java

```diff
@@ -64,6 +72,13 @@ public void processingLatency(
         .observe((processed - written) / 1000f);
   }
 
+  public void processingDuration(
+      final RecordType recordType, final long started, final long processed) {
+    PROCESSING_DURATION
+        .labels(recordType.name(), partitionIdLabel)
```

Comment:
I think it would be interesting to find hotspots by adding value type/intent as labels - however I'm not sure if that creates too much dimensions/data explosion. We can definitely do that as a second iteration though.

Reply:
Yes. We can improve it when we need it.